pricelist_cache/models/product_pricelist_item.py

```diff
@@ -25,6 +27,19 @@ def _has_date_range(self):
         """Returns whether any of the item records in recordset is based on dates."""
         return any(bool(record.date_start or record.date_end) for record in self)
 
+    def _get_pricelist_products(self):
+        products = []
+        for rec in self:
+            if rec.product_tmpl_id.id:
+                products = (
+                    self.env["product.product"]
+                    .search([("product_tmpl_id", "=", rec.product_tmpl_id.id)])
+                    .ids
+                )
+            else:
+                products = rec.product_id.ids
+        return products
+
```

Comment:
This should be called _ids if it is returning ids.
Why cant we just use `rec.rec.product_tmpl_id.product_variant_ids.ids` instead of search?

```suggestion
    def _get_pricelist_product_ids(self):
        product_ids = []
        for rec in self:
            if rec.product_tmpl_id.id:
                product_ids = rec.product_tmpl_id.product_variant_ids.ids
            else:
                product_ids = rec.product_id.ids
        return product_ids

```